github/repos.go

```diff
@@ -2402,3 +2402,35 @@ func (s *RepositoriesService) DisablePrivateReporting(ctx context.Context, owner
 
 	return resp, nil
 }
+
+// CheckPrivateReporting represents whether private vulnerability reporting is enabled.
+type CheckPrivateReporting struct {
+	Enabled bool `json:"enabled,omitempty"`
+}
+
+// IsPrivateReportingEnabled checks if private vulnerability reporting is enabled
+// for the repository and returns a boolean indicating the status.
+//
+// GitHub API docs: https://docs.github.com/rest/repos/repos#check-if-private-vulnerability-reporting-is-enabled-for-a-repository
+//
+//meta:operation GET /repos/{owner}/{repo}/private-vulnerability-reporting
+func (s *RepositoriesService) IsPrivateReportingEnabled(ctx context.Context, owner, repo string) (bool, *Response, error) {
+	u := fmt.Sprintf("repos/%v/%v/private-vulnerability-reporting", owner, repo)
+
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return false, nil, err
+	}
+
+	privateReporting := new(CheckPrivateReporting)
+	resp, err := s.client.Do(ctx, req, privateReporting)
+	if privateReporting.Enabled {
```

Comment:
We don't need to explicitly check for `StatusNotFound` below as far as I can tell from the docs, so this could all be simplified to just:

```suggestion
	return privateReporting.Enabled, resp, err
```

Reply:
The only reason I called this out here is because the official documentation doesn't say that it ever returns a "404 Not Found"... hence I don't think we need it in this particular case.